velox/exec/Task.cpp

```diff
@@ -168,6 +168,10 @@ Task::~Task() {
   } catch (const std::exception& e) {
     LOG(WARNING) << "Caught exception in ~Task(): " << e.what();
   }
+  // NOTE: this is a hack to enforce destruction on 'planFragment_'. We found in
```

Comment:
@laithsakka Laith, curious if you'd have an explanation for this behavior.

Reply:
Sounds like a cyclic dependency but `planFragment_` is not shared...

Reply:
@Yuhta Thanks for the suggestion. I am curious if it is std::shared_ptr<folly::Executor> executor_ in QueryCtx destruction hangs at executor_ pool destruction. Once it happened, it is always triggered by the lambda function callback which is executed by the driver executor. Will try to verify.